### PR TITLE
Add contributing guidelines and mention in README.

### PR DESCRIPTION
**Pull request by BlitzKraft**

### CODE REVIEW

**Comment by BlitzKraft** [comment]
Fix #6

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    disable:
+      - too-many-ancestors
+      - too-few-public-methods
+      - invalid-unary-operand-type
+```
+
+This is a style **guide**, not rules. There will be exceptions and use your
+best judgement when you need to.
+
+As a quick reference, the above rules are explained below.
```
Comment (AKuederle) [inline CONTRIBUTING.md]: I think it is not really clear that the rules listed above are the ones that are disabled in the linter and not the one that are enabled

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  run: false
+  options:
+    max-line-length: 120
+    disable:
+      - too-many-ancestors
+      - too-few-public-methods
+      - invalid-unary-operand-type
+```
+
+This is a style **guide**, not rules. There will be exceptions and use your
```
Comment (AKuederle) [inline CONTRIBUTING.md]: Mmh actually I was planning to enforce some of the guidelines via a CI linter (#3 ) . But it should be fine for now. I will add the relevant information, once we have that actually running

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+* After tests and bug fixes, it will be merged into master for release.
+
+The reason for this is that `master` will always be ready for end users and be
+stable, without any unexpected bugs and issues (hopefully). `devel` is for
+continuous testing and development and to catch bugs before they reach the end
+user. Everything else is to keep the code history nice and traceable. 
+
+## Some gotchas
+
+* Before you start working on an issue or feature, check your local repo is up
```
Comment (AKuederle) [inline CONTRIBUTING.md]: Typo: check *if* your

File: CONTRIBUTING.md (hunk 1 above)
Comment (BlitzKraft) [reply CONTRIBUTING.md]: There was disable right above them, and thought mentioning it again would be too much.

Comment (AKuederle) [reply CONTRIBUTING.md]: Ok. Then we just leave it like that

**Pull request merged**